hydra-cluster/src/CardanoNode.hs

```diff
@@ -187,9 +198,18 @@ withCardanoNodeOnKnownNetwork ::
 withCardanoNodeOnKnownNetwork tracer workDir knownNetwork action = do
   copyKnownNetworkFiles
   networkId <- readNetworkId
-  withCardanoNode tracer networkId workDir args $ \node -> do
+  withCardanoNode tracer workDir args $ \nodeSocket -> do
     traceWith tracer MsgNodeIsReady
-    action node
+    pparams <- queryProtocolParameters networkId nodeSocket QueryTip
```

Comment:
Seems like there is an opportunity for removing some duplication here

Reply:
I saw it but since we need to go from the cardano-api to ledger `PParams` here we would need to handle  `Either` and I decided to just leave it as is since this is test code.